workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -158,11 +160,22 @@ def _restore_amqp_vhosts_and_users(self):
     def _restore_certificate(self):
         archive_cert_dir = os.path.join(self._tempdir, ARCHIVE_CERT_DIR)
         old_cert_dir = os.path.dirname(get_local_rest_certificate())
-        new_cert_dir = old_cert_dir + '_from_snapshot'
-        utils.copy_snapshot_path(archive_cert_dir, new_cert_dir)
+        new_cert_dir = '{0}_from_snapshot_{1}'.format(old_cert_dir,
+                                                      self._snapshot_id)
+        old_cert = os.path.join(old_cert_dir, INTERNAL_CERT_FILENAME)
+        old_key = os.path.join(old_cert_dir, INTERNAL_KEY_FILENAME)
+        new_cert = os.path.join(new_cert_dir, INTERNAL_CERT_FILENAME)
+        new_key = os.path.join(new_cert_dir, INTERNAL_KEY_FILENAME)
         time_to_wait_for_workflow_to_finish = 3
-        cmd = 'sleep {0}; rm -rf {1}; mv {2} {1}'.format(
-            time_to_wait_for_workflow_to_finish, old_cert_dir, new_cert_dir)
+        cmd = 'sleep {0}; rm -rf {1} {2}; mv {3} {1}; mv {4} {2}'.format(
```

Comment:
Let's replace  `{1}` with `{old_cert}` and, `{2}` with `{old_key}` and so on, to make this more readable.